File: common/src/main/java/dev/mayaqq/estrogen/registry/EstrogenPotions.java
package dev.mayaqq.estrogen.registry;

import dev.mayaqq.estrogen.Estrogen;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.effect.MobEffectInstance;
import net.minecraft.world.item.alchemy.Potion;
import uwu.serenity.critter.api.entry.RegistryEntry;
import uwu.serenity.critter.api.generic.Registrar;

public class EstrogenPotions {
    public static final Registrar<Potion> POTIONS = Estrogen.REGISTRIES.getRegistrar(Registries.POTION);

    public static final RegistryEntry<Potion> ESTROGEN_POTION = POTIONS.entry("estrogen", () ->
            new Potion("estrogen", new MobEffectInstance(EstrogenEffects.ESTROGEN_EFFECT.get(), 12000))
    ).register();
}

File: src/main/kotlin/dev/mayaqq/estrogen/content/EstrogenPotions.kt
package dev.mayaqq.estrogen.content

import dev.mayaqq.estrogen.MOD_ID
import invoke.kitty.kritter.registry.api.Registrar
import invoke.kitty.kritter.registry.api.builder.entry
import invoke.kitty.kritter.registry.api.entry.holder
import net.minecraft.core.registries.Registries
import net.minecraft.world.effect.MobEffectInstance
import net.minecraft.world.item.alchemy.Potion

object EstrogenPotions : Registrar<Potion> by Registrar(MOD_ID, Registries.POTION) {
    val EstrogenPotion = entry("estrogen_potion", {Potion("estrogen", MobEffectInstance(EstrogenEffects.Estrogen.holder, 12000))}) {}
}